pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -601,17 +655,49 @@ func getSpecEnv(envs []corev1.EnvVar, name string) string {
 	return ""
 }
 
-func getSmartRouterProtocol(cr *api.KieApp) string {
+func getSmartRouterTermination(cr *api.KieApp) (termination string, protocol string) {
 	if cr.Status.Applied.Objects.SmartRouter != nil {
-		if len(cr.Status.Applied.Objects.SmartRouter.Protocol) == 0 && cr.Status.Applied.CommonConfig.DisableSsl {
-			return constants.HttpProtocol
-		} else if len(cr.Status.Applied.Objects.SmartRouter.Protocol) == 0 && !cr.Status.Applied.CommonConfig.DisableSsl {
-			return constants.HttpsProtocol
+		if cr.Status.Applied.Objects.SmartRouter.TerminationRoute.EnableEdge {
+			// Edge is always http
+			return constants.EdgeTermination, constants.HttpProtocol
+		} else if cr.Status.Applied.CommonConfig.DisableSsl {
+			return constants.DefaultTermination, constants.HttpProtocol
+		} else {
+			return constants.DefaultTermination, constants.HttpsProtocol
+		}
+	} else {
+		return constants.DefaultTermination, constants.HttpsProtocol
+	}
+}
+
+func getConsoleTermination(cr *api.KieApp) (termination string, protocol string) {
+	if cr.Status.Applied.Objects.Console.TerminationRoute != nil {
+		if cr.Status.Applied.Objects.Console.TerminationRoute.EnableEdge {
+			// Edge is always http
+			return constants.EdgeTermination, constants.HttpProtocol
+		} else if cr.Spec.CommonConfig.DisableSsl {
+			return constants.DefaultTermination, constants.HttpProtocol
 		} else {
-			return cr.Status.Applied.Objects.SmartRouter.Protocol
+			return constants.DefaultTermination, constants.HttpsProtocol
 		}
+	} else {
+		return constants.DefaultTermination, constants.HttpsProtocol
+	}
+}
+
+func getServerTermination(server *api.KieServerSet, cr *api.KieApp) (termination string, protocol string) {
+	if server.TerminationRoute != nil {
+		if len(server.TerminationRoute.Protocol) > 0 {
+			// Edge is always http
```

Comment:
This check looks wrong, there is check just some len of the parameter and we do not care what is configured there.

Reply:
Was a old behaviour now we check if enableEdge value